src/LandmarkParser.java

```diff
@@ -0,0 +1,38 @@
+
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.json.simple.parser.ParseException;
+
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.HashMap;
+
+public class LandmarkParser {
+
+
+  public static HashMap<Integer, Landmark> parseLandmarks() {
+    HashMap<Integer, Landmark> list = new HashMap<>();
+    JSONParser parser = new JSONParser();
+
+    File file = new File("resources/landmarks/landmarks.json");
+
+    JSONObject obj = null;
+
+    try {
+      obj = (JSONObject) parser.parse(new FileReader(file));
+      for(Object i : obj.values()){
+        Landmark lm = new Landmark((JSONObject) i);
+        list.put(Integer.parseInt(String.valueOf(lm.id)), lm);
```

Comment:
First a minor point, strings aren't a great way to cast integers. I would recommend something like:
```
int intValue = Math.toIntExact(longValue); // Java 8+ solution
```
More info in this [link](https://stackoverflow.com/questions/4355303/how-can-i-convert-a-long-to-int-in-java).

The more important point is that you shouldn't be mixing `int` with `long`, so you should either only use longs or convert the id from a `long` to an `int` in the very beginning (`Landmark` constructor).

Reply:
I fixed the way the data was imported and utilized the Utility class Cast to int function to fix the way the ID was cast.